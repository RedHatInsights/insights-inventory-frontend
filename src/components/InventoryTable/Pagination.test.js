/* eslint-disable react/prop-types */
/* eslint-disable camelcase */
import React from 'react';
import Pagination from './Pagination';
import configureStore from 'redux-mock-store';
import { createPromise as promiseMiddleware } from 'redux-promise-middleware';
import { Provider } from 'react-redux';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import userEvent from '@testing-library/user-event';
jest.mock('../../store/actions', () => {
  const actions = jest.requireActual('../../store/actions');
  const { ACTION_TYPES } = jest.requireActual('../../store/action-types');
  return {
    __esModule: true,
    ...actions,
    loadEntities: () => ({
      type: ACTION_TYPES.LOAD_ENTITIES,
      payload: () => Promise.resolve({}),
      meta: { showTags: undefined },
    }),
  };
});

describe('Pagination', () => {
  let initialState;
  let mockStore;
  let onRefreshData;

  let WrappedPagination;

  beforeEach(() => {
    mockStore = configureStore([promiseMiddleware()]);
    initialState = {
      entities: {
        activeFilters: [{}],
        loaded: true,
      },
    };
    onRefreshData = jest
      .fn()
      .mockImplementation(() => Promise.resolve({ results: [], total: 0 }));

    // eslint-disable-next-line react/display-name
    WrappedPagination = ({ store, ...props }) => (
      <Provider store={store}>
        <Pagination loaded {...props} onRefreshData={onRefreshData} />
      </Provider>
    );
  });

  describe('render', () => {
    it('should render correctly - no data', () => {
      const store = mockStore({ entities: {} });
      const view = render(<WrappedPagination store={store} loaded={false} />);

      expect(view.asFragment()).toMatchSnapshot();
    });

    it('should render correctly with data', () => {
      const store = mockStore(initialState);
      const view = render(<WrappedPagination store={store} />);

      expect(view.asFragment()).toMatchSnapshot();
      expect(
        screen.getByRole('button', {
          name: /go to first page/i,
        })
      ).toBeDisabled();
      expect(
        screen.getByRole('button', {
          name: /go to previous page/i,
        })
      ).toBeDisabled();
      expect(
        screen.getByRole('button', {
          name: /go to next page/i,
        })
      ).toBeDisabled();
      expect(
        screen.getByRole('button', {
          name: /go to last page/i,
        })
      ).toBeDisabled();
    });

    it('should render correctly - with no access', () => {
      const store = mockStore(initialState);
      const view = render(
        <WrappedPagination store={store} hasAccess={false} />
      );

      expect(view.asFragment()).toMatchSnapshot();
    });

    it('should render correctly with data and props', () => {
      const store = mockStore(initialState);
      const view = render(
        <WrappedPagination store={store} page={1} perPage={50} total={500} />
      );

      expect(view.asFragment()).toMatchSnapshot();
      expect(
        screen.getByRole('button', {
          name: /go to first page/i,
        })
      ).toBeDisabled();
      expect(
        screen.getByRole('button', {
          name: /go to previous page/i,
        })
      ).toBeDisabled();
    });

    it('should render correctly with data and isFull', () => {
      const store = mockStore(initialState);
      const view = render(
        <WrappedPagination
          store={store}
          page={1}
          perPage={50}
          total={500}
          isFull
        />
      );

      expect(view.asFragment()).toMatchSnapshot();
      expect(
        screen.getByRole('button', {
          name: /go to first page/i,
        })
      ).toBeDisabled();
      expect(
        screen.getByRole('button', {
          name: /go to previous page/i,
        })
      ).toBeDisabled();
    });
  });

  describe('API', () => {
    it('should call perPage change', async () => {
      const store = mockStore(initialState);
      render(
        <WrappedPagination store={store} page={1} perPage={50} total={500} />
      );

      await userEvent.click(
        screen.getByRole('button', {
          name: /items per page/i,
        })
      );
      await userEvent.click(
        screen.getByRole('menuitem', {
          name: /10 per page/i,
        })
      );
      expect(onRefreshData).toHaveBeenCalledWith({ page: 1, per_page: 10 });
    });

    it('should call onSetPage change without', async () => {
      const store = mockStore(initialState);
      render(
        <WrappedPagination store={store} page={1} perPage={50} total={500} />
      );

      await userEvent.click(
        screen.getByRole('button', {
          name: /go to next page/i,
        })
      );
      expect(onRefreshData).toHaveBeenCalledWith({ page: 2 });
    });
  });
});
