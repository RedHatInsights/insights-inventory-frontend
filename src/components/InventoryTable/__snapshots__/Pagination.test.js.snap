// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination render should render correctly - no data 1`] = `<DocumentFragment />`;

exports[`Pagination render should render correctly - with no access 1`] = `
<DocumentFragment>
  <div
    class="pf-v5-c-pagination"
    data-ouia-component-id="OUIA-Generated-Pagination-top-2"
    data-ouia-component-type="PF5/Pagination"
    data-ouia-safe="true"
    id="options-menu-top-pagination"
    style="--pf-v5-c-pagination__nav-page-select--c-form-control--width-chars: 2;"
  >
    <div
      class="pf-v5-c-pagination__total-items"
    >
      <b>
        0 - 0
      </b>
       of 
      <b>
        0
      </b>
       
    </div>
    <button
      aria-expanded="false"
      aria-haspopup="listbox"
      aria-label="Items per page"
      class="pf-v5-c-menu-toggle pf-m-plain pf-m-text pf-m-disabled"
      data-ouia-component-id="OUIA-Generated-MenuToggle-plainText-2"
      data-ouia-component-type="PF5/MenuToggle"
      data-ouia-safe="true"
      disabled=""
      id="options-menu-top-toggle"
      type="button"
    >
      <span
        class="pf-v5-c-menu-toggle__text"
      >
        <b>
          0 - 0
        </b>
         of 
        <b>
          0
        </b>
         
      </span>
      <span
        class="pf-v5-c-menu-toggle__controls"
      >
        <span
          class="pf-v5-c-menu-toggle__toggle-icon"
        >
          <svg
            aria-hidden="true"
            class="pf-v5-svg"
            fill="currentColor"
            height="1em"
            role="img"
            viewBox="0 0 320 512"
            width="1em"
          >
            <path
              d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
            />
          </svg>
        </span>
      </span>
    </button>
    <nav
      aria-label="Pagination"
      class="pf-v5-c-pagination__nav"
    >
      <div
        class="pf-v5-c-pagination__nav-control pf-m-first"
      >
        <button
          aria-disabled="true"
          aria-label="Go to first page"
          class="pf-v5-c-button pf-m-plain pf-m-disabled"
          data-action="first"
          data-ouia-component-id="OUIA-Generated-Button-plain-5"
          data-ouia-component-type="PF5/Button"
          data-ouia-safe="true"
          disabled=""
          type="button"
        >
          <svg
            aria-hidden="true"
            class="pf-v5-svg"
            fill="currentColor"
            height="1em"
            role="img"
            viewBox="0 0 448 512"
            width="1em"
          >
            <path
              d="M223.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L319.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L393.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34zm-192 34l136 136c9.4 9.4 24.6 9.4 33.9 0l22.6-22.6c9.4-9.4 9.4-24.6 0-33.9L127.9 256l96.4-96.4c9.4-9.4 9.4-24.6 0-33.9L201.7 103c-9.4-9.4-24.6-9.4-33.9 0l-136 136c-9.5 9.4-9.5 24.6-.1 34z"
            />
          </svg>
        </button>
      </div>
      <div
        class="pf-v5-c-pagination__nav-control"
      >
        <button
          aria-disabled="true"
          aria-label="Go to previous page"
          class="pf-v5-c-button pf-m-plain pf-m-disabled"
          data-action="previous"
          data-ouia-component-id="OUIA-Generated-Button-plain-6"
          data-ouia-component-type="PF5/Button"
          data-ouia-safe="true"
          disabled=""
          type="button"
        >
          <svg
            aria-hidden="true"
            class="pf-v5-svg"
            fill="currentColor"
            height="1em"
            role="img"
            viewBox="0 0 256 512"
            width="1em"
          >
            <path
              d="M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z"
            />
          </svg>
        </button>
      </div>
      <div
        class="pf-v5-c-pagination__nav-page-select"
      >
        <span
          class="pf-v5-c-form-control pf-m-disabled"
        >
          <input
            aria-invalid="false"
            aria-label="Current page"
            data-ouia-component-id="OUIA-Generated-TextInputBase-2"
            data-ouia-component-type="PF5/TextInput"
            data-ouia-safe="true"
            disabled=""
            max="0"
            min="1"
            type="number"
            value="0"
          />
        </span>
        <span
          aria-hidden="true"
        >
          of 0
        </span>
      </div>
      <div
        class="pf-v5-c-pagination__nav-control"
      >
        <button
          aria-disabled="true"
          aria-label="Go to next page"
          class="pf-v5-c-button pf-m-plain pf-m-disabled"
          data-action="next"
          data-ouia-component-id="OUIA-Generated-Button-plain-7"
          data-ouia-component-type="PF5/Button"
          data-ouia-safe="true"
          disabled=""
          type="button"
        >
          <svg
            aria-hidden="true"
            class="pf-v5-svg"
            fill="currentColor"
            height="1em"
            role="img"
            viewBox="0 0 256 512"
            width="1em"
          >
            <path
              d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
            />
          </svg>
        </button>
      </div>
      <div
        class="pf-v5-c-pagination__nav-control pf-m-last"
      >
        <button
          aria-disabled="true"
          aria-label="Go to last page"
          class="pf-v5-c-button pf-m-plain pf-m-disabled"
          data-action="last"
          data-ouia-component-id="OUIA-Generated-Button-plain-8"
          data-ouia-component-type="PF5/Button"
          data-ouia-safe="true"
          disabled=""
          type="button"
        >
          <svg
            aria-hidden="true"
            class="pf-v5-svg"
            fill="currentColor"
            height="1em"
            role="img"
            viewBox="0 0 448 512"
            width="1em"
          >
            <path
              d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34zm192-34l-136-136c-9.4-9.4-24.6-9.4-33.9 0l-22.6 22.6c-9.4 9.4-9.4 24.6 0 33.9l96.4 96.4-96.4 96.4c-9.4 9.4-9.4 24.6 0 33.9l22.6 22.6c9.4 9.4 24.6 9.4 33.9 0l136-136c9.4-9.2 9.4-24.4 0-33.8z"
            />
          </svg>
        </button>
      </div>
    </nav>
  </div>
</DocumentFragment>
`;

exports[`Pagination render should render correctly with data 1`] = `
<DocumentFragment>
  <div
    class="pf-v5-c-pagination"
    data-ouia-component-id="OUIA-Generated-Pagination-top-1"
    data-ouia-component-type="PF5/Pagination"
    data-ouia-safe="true"
    id="options-menu-top-pagination"
    style="--pf-v5-c-pagination__nav-page-select--c-form-control--width-chars: 2;"
  >
    <div
      class="pf-v5-c-pagination__total-items"
    >
      <b>
        0 - 0
      </b>
       of 
      <b>
        0
      </b>
       
    </div>
    <button
      aria-expanded="false"
      aria-haspopup="listbox"
      aria-label="Items per page"
      class="pf-v5-c-menu-toggle pf-m-plain pf-m-text"
      data-ouia-component-id="OUIA-Generated-MenuToggle-plainText-1"
      data-ouia-component-type="PF5/MenuToggle"
      data-ouia-safe="true"
      id="options-menu-top-toggle"
      type="button"
    >
      <span
        class="pf-v5-c-menu-toggle__text"
      >
        <b>
          0 - 0
        </b>
         of 
        <b>
          0
        </b>
         
      </span>
      <span
        class="pf-v5-c-menu-toggle__controls"
      >
        <span
          class="pf-v5-c-menu-toggle__toggle-icon"
        >
          <svg
            aria-hidden="true"
            class="pf-v5-svg"
            fill="currentColor"
            height="1em"
            role="img"
            viewBox="0 0 320 512"
            width="1em"
          >
            <path
              d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
            />
          </svg>
        </span>
      </span>
    </button>
    <nav
      aria-label="Pagination"
      class="pf-v5-c-pagination__nav"
    >
      <div
        class="pf-v5-c-pagination__nav-control pf-m-first"
      >
        <button
          aria-disabled="true"
          aria-label="Go to first page"
          class="pf-v5-c-button pf-m-plain pf-m-disabled"
          data-action="first"
          data-ouia-component-id="OUIA-Generated-Button-plain-1"
          data-ouia-component-type="PF5/Button"
          data-ouia-safe="true"
          disabled=""
          type="button"
        >
          <svg
            aria-hidden="true"
            class="pf-v5-svg"
            fill="currentColor"
            height="1em"
            role="img"
            viewBox="0 0 448 512"
            width="1em"
          >
            <path
              d="M223.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L319.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L393.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34zm-192 34l136 136c9.4 9.4 24.6 9.4 33.9 0l22.6-22.6c9.4-9.4 9.4-24.6 0-33.9L127.9 256l96.4-96.4c9.4-9.4 9.4-24.6 0-33.9L201.7 103c-9.4-9.4-24.6-9.4-33.9 0l-136 136c-9.5 9.4-9.5 24.6-.1 34z"
            />
          </svg>
        </button>
      </div>
      <div
        class="pf-v5-c-pagination__nav-control"
      >
        <button
          aria-disabled="true"
          aria-label="Go to previous page"
          class="pf-v5-c-button pf-m-plain pf-m-disabled"
          data-action="previous"
          data-ouia-component-id="OUIA-Generated-Button-plain-2"
          data-ouia-component-type="PF5/Button"
          data-ouia-safe="true"
          disabled=""
          type="button"
        >
          <svg
            aria-hidden="true"
            class="pf-v5-svg"
            fill="currentColor"
            height="1em"
            role="img"
            viewBox="0 0 256 512"
            width="1em"
          >
            <path
              d="M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z"
            />
          </svg>
        </button>
      </div>
      <div
        class="pf-v5-c-pagination__nav-page-select"
      >
        <span
          class="pf-v5-c-form-control pf-m-disabled"
        >
          <input
            aria-invalid="false"
            aria-label="Current page"
            data-ouia-component-id="OUIA-Generated-TextInputBase-1"
            data-ouia-component-type="PF5/TextInput"
            data-ouia-safe="true"
            disabled=""
            max="0"
            min="1"
            type="number"
            value="0"
          />
        </span>
        <span
          aria-hidden="true"
        >
          of 0
        </span>
      </div>
      <div
        class="pf-v5-c-pagination__nav-control"
      >
        <button
          aria-disabled="true"
          aria-label="Go to next page"
          class="pf-v5-c-button pf-m-plain pf-m-disabled"
          data-action="next"
          data-ouia-component-id="OUIA-Generated-Button-plain-3"
          data-ouia-component-type="PF5/Button"
          data-ouia-safe="true"
          disabled=""
          type="button"
        >
          <svg
            aria-hidden="true"
            class="pf-v5-svg"
            fill="currentColor"
            height="1em"
            role="img"
            viewBox="0 0 256 512"
            width="1em"
          >
            <path
              d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
            />
          </svg>
        </button>
      </div>
      <div
        class="pf-v5-c-pagination__nav-control pf-m-last"
      >
        <button
          aria-disabled="true"
          aria-label="Go to last page"
          class="pf-v5-c-button pf-m-plain pf-m-disabled"
          data-action="last"
          data-ouia-component-id="OUIA-Generated-Button-plain-4"
          data-ouia-component-type="PF5/Button"
          data-ouia-safe="true"
          disabled=""
          type="button"
        >
          <svg
            aria-hidden="true"
            class="pf-v5-svg"
            fill="currentColor"
            height="1em"
            role="img"
            viewBox="0 0 448 512"
            width="1em"
          >
            <path
              d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34zm192-34l-136-136c-9.4-9.4-24.6-9.4-33.9 0l-22.6 22.6c-9.4 9.4-9.4 24.6 0 33.9l96.4 96.4-96.4 96.4c-9.4 9.4-9.4 24.6 0 33.9l22.6 22.6c9.4 9.4 24.6 9.4 33.9 0l136-136c9.4-9.2 9.4-24.4 0-33.8z"
            />
          </svg>
        </button>
      </div>
    </nav>
  </div>
</DocumentFragment>
`;

exports[`Pagination render should render correctly with data and isFull 1`] = `
<DocumentFragment>
  <div
    class="pf-v5-c-pagination pf-m-bottom"
    data-ouia-component-id="OUIA-Generated-Pagination-bottom-1"
    data-ouia-component-type="PF5/Pagination"
    data-ouia-safe="true"
    id="options-menu-bottom-pagination"
    style="--pf-v5-c-pagination__nav-page-select--c-form-control--width-chars: 2;"
  >
    <button
      aria-expanded="false"
      aria-haspopup="listbox"
      aria-label="Items per page"
      class="pf-v5-c-menu-toggle pf-m-plain pf-m-text"
      data-ouia-component-id="OUIA-Generated-MenuToggle-plainText-4"
      data-ouia-component-type="PF5/MenuToggle"
      data-ouia-safe="true"
      id="options-menu-bottom-toggle"
      type="button"
    >
      <span
        class="pf-v5-c-menu-toggle__text"
      >
        <b>
          1 - 50
        </b>
         of 
        <b>
          500
        </b>
         
      </span>
      <span
        class="pf-v5-c-menu-toggle__controls"
      >
        <span
          class="pf-v5-c-menu-toggle__toggle-icon"
        >
          <svg
            aria-hidden="true"
            class="pf-v5-svg"
            fill="currentColor"
            height="1em"
            role="img"
            viewBox="0 0 320 512"
            width="1em"
          >
            <path
              d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
            />
          </svg>
        </span>
      </span>
    </button>
    <nav
      aria-label="Pagination"
      class="pf-v5-c-pagination__nav"
    >
      <div
        class="pf-v5-c-pagination__nav-control pf-m-first"
      >
        <button
          aria-disabled="true"
          aria-label="Go to first page"
          class="pf-v5-c-button pf-m-plain pf-m-disabled"
          data-action="first"
          data-ouia-component-id="OUIA-Generated-Button-plain-13"
          data-ouia-component-type="PF5/Button"
          data-ouia-safe="true"
          disabled=""
          type="button"
        >
          <svg
            aria-hidden="true"
            class="pf-v5-svg"
            fill="currentColor"
            height="1em"
            role="img"
            viewBox="0 0 448 512"
            width="1em"
          >
            <path
              d="M223.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L319.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L393.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34zm-192 34l136 136c9.4 9.4 24.6 9.4 33.9 0l22.6-22.6c9.4-9.4 9.4-24.6 0-33.9L127.9 256l96.4-96.4c9.4-9.4 9.4-24.6 0-33.9L201.7 103c-9.4-9.4-24.6-9.4-33.9 0l-136 136c-9.5 9.4-9.5 24.6-.1 34z"
            />
          </svg>
        </button>
      </div>
      <div
        class="pf-v5-c-pagination__nav-control"
      >
        <button
          aria-disabled="true"
          aria-label="Go to previous page"
          class="pf-v5-c-button pf-m-plain pf-m-disabled"
          data-action="previous"
          data-ouia-component-id="OUIA-Generated-Button-plain-14"
          data-ouia-component-type="PF5/Button"
          data-ouia-safe="true"
          disabled=""
          type="button"
        >
          <svg
            aria-hidden="true"
            class="pf-v5-svg"
            fill="currentColor"
            height="1em"
            role="img"
            viewBox="0 0 256 512"
            width="1em"
          >
            <path
              d="M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z"
            />
          </svg>
        </button>
      </div>
      <div
        class="pf-v5-c-pagination__nav-page-select"
      >
        <span
          class="pf-v5-c-form-control"
        >
          <input
            aria-invalid="false"
            aria-label="Current page"
            data-ouia-component-id="OUIA-Generated-TextInputBase-4"
            data-ouia-component-type="PF5/TextInput"
            data-ouia-safe="true"
            max="10"
            min="1"
            type="number"
            value="1"
          />
        </span>
        <span
          aria-hidden="true"
        >
          of 10
        </span>
      </div>
      <div
        class="pf-v5-c-pagination__nav-control"
      >
        <button
          aria-disabled="false"
          aria-label="Go to next page"
          class="pf-v5-c-button pf-m-plain"
          data-action="next"
          data-ouia-component-id="OUIA-Generated-Button-plain-15"
          data-ouia-component-type="PF5/Button"
          data-ouia-safe="true"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="pf-v5-svg"
            fill="currentColor"
            height="1em"
            role="img"
            viewBox="0 0 256 512"
            width="1em"
          >
            <path
              d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
            />
          </svg>
        </button>
      </div>
      <div
        class="pf-v5-c-pagination__nav-control pf-m-last"
      >
        <button
          aria-disabled="false"
          aria-label="Go to last page"
          class="pf-v5-c-button pf-m-plain"
          data-action="last"
          data-ouia-component-id="OUIA-Generated-Button-plain-16"
          data-ouia-component-type="PF5/Button"
          data-ouia-safe="true"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="pf-v5-svg"
            fill="currentColor"
            height="1em"
            role="img"
            viewBox="0 0 448 512"
            width="1em"
          >
            <path
              d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34zm192-34l-136-136c-9.4-9.4-24.6-9.4-33.9 0l-22.6 22.6c-9.4 9.4-9.4 24.6 0 33.9l96.4 96.4-96.4 96.4c-9.4 9.4-9.4 24.6 0 33.9l22.6 22.6c9.4 9.4 24.6 9.4 33.9 0l136-136c9.4-9.2 9.4-24.4 0-33.8z"
            />
          </svg>
        </button>
      </div>
    </nav>
  </div>
</DocumentFragment>
`;

exports[`Pagination render should render correctly with data and props 1`] = `
<DocumentFragment>
  <div
    class="pf-v5-c-pagination"
    data-ouia-component-id="OUIA-Generated-Pagination-top-3"
    data-ouia-component-type="PF5/Pagination"
    data-ouia-safe="true"
    id="options-menu-top-pagination"
    style="--pf-v5-c-pagination__nav-page-select--c-form-control--width-chars: 2;"
  >
    <div
      class="pf-v5-c-pagination__total-items"
    >
      <b>
        1 - 50
      </b>
       of 
      <b>
        500
      </b>
       
    </div>
    <button
      aria-expanded="false"
      aria-haspopup="listbox"
      aria-label="Items per page"
      class="pf-v5-c-menu-toggle pf-m-plain pf-m-text"
      data-ouia-component-id="OUIA-Generated-MenuToggle-plainText-3"
      data-ouia-component-type="PF5/MenuToggle"
      data-ouia-safe="true"
      id="options-menu-top-toggle"
      type="button"
    >
      <span
        class="pf-v5-c-menu-toggle__text"
      >
        <b>
          1 - 50
        </b>
         of 
        <b>
          500
        </b>
         
      </span>
      <span
        class="pf-v5-c-menu-toggle__controls"
      >
        <span
          class="pf-v5-c-menu-toggle__toggle-icon"
        >
          <svg
            aria-hidden="true"
            class="pf-v5-svg"
            fill="currentColor"
            height="1em"
            role="img"
            viewBox="0 0 320 512"
            width="1em"
          >
            <path
              d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
            />
          </svg>
        </span>
      </span>
    </button>
    <nav
      aria-label="Pagination"
      class="pf-v5-c-pagination__nav"
    >
      <div
        class="pf-v5-c-pagination__nav-control pf-m-first"
      >
        <button
          aria-disabled="true"
          aria-label="Go to first page"
          class="pf-v5-c-button pf-m-plain pf-m-disabled"
          data-action="first"
          data-ouia-component-id="OUIA-Generated-Button-plain-9"
          data-ouia-component-type="PF5/Button"
          data-ouia-safe="true"
          disabled=""
          type="button"
        >
          <svg
            aria-hidden="true"
            class="pf-v5-svg"
            fill="currentColor"
            height="1em"
            role="img"
            viewBox="0 0 448 512"
            width="1em"
          >
            <path
              d="M223.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L319.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L393.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34zm-192 34l136 136c9.4 9.4 24.6 9.4 33.9 0l22.6-22.6c9.4-9.4 9.4-24.6 0-33.9L127.9 256l96.4-96.4c9.4-9.4 9.4-24.6 0-33.9L201.7 103c-9.4-9.4-24.6-9.4-33.9 0l-136 136c-9.5 9.4-9.5 24.6-.1 34z"
            />
          </svg>
        </button>
      </div>
      <div
        class="pf-v5-c-pagination__nav-control"
      >
        <button
          aria-disabled="true"
          aria-label="Go to previous page"
          class="pf-v5-c-button pf-m-plain pf-m-disabled"
          data-action="previous"
          data-ouia-component-id="OUIA-Generated-Button-plain-10"
          data-ouia-component-type="PF5/Button"
          data-ouia-safe="true"
          disabled=""
          type="button"
        >
          <svg
            aria-hidden="true"
            class="pf-v5-svg"
            fill="currentColor"
            height="1em"
            role="img"
            viewBox="0 0 256 512"
            width="1em"
          >
            <path
              d="M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z"
            />
          </svg>
        </button>
      </div>
      <div
        class="pf-v5-c-pagination__nav-page-select"
      >
        <span
          class="pf-v5-c-form-control"
        >
          <input
            aria-invalid="false"
            aria-label="Current page"
            data-ouia-component-id="OUIA-Generated-TextInputBase-3"
            data-ouia-component-type="PF5/TextInput"
            data-ouia-safe="true"
            max="10"
            min="1"
            type="number"
            value="1"
          />
        </span>
        <span
          aria-hidden="true"
        >
          of 10
        </span>
      </div>
      <div
        class="pf-v5-c-pagination__nav-control"
      >
        <button
          aria-disabled="false"
          aria-label="Go to next page"
          class="pf-v5-c-button pf-m-plain"
          data-action="next"
          data-ouia-component-id="OUIA-Generated-Button-plain-11"
          data-ouia-component-type="PF5/Button"
          data-ouia-safe="true"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="pf-v5-svg"
            fill="currentColor"
            height="1em"
            role="img"
            viewBox="0 0 256 512"
            width="1em"
          >
            <path
              d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
            />
          </svg>
        </button>
      </div>
      <div
        class="pf-v5-c-pagination__nav-control pf-m-last"
      >
        <button
          aria-disabled="false"
          aria-label="Go to last page"
          class="pf-v5-c-button pf-m-plain"
          data-action="last"
          data-ouia-component-id="OUIA-Generated-Button-plain-12"
          data-ouia-component-type="PF5/Button"
          data-ouia-safe="true"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="pf-v5-svg"
            fill="currentColor"
            height="1em"
            role="img"
            viewBox="0 0 448 512"
            width="1em"
          >
            <path
              d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34zm192-34l-136-136c-9.4-9.4-24.6-9.4-33.9 0l-22.6 22.6c-9.4 9.4-9.4 24.6 0 33.9l96.4 96.4-96.4 96.4c-9.4 9.4-9.4 24.6 0 33.9l22.6 22.6c9.4 9.4 24.6 9.4 33.9 0l136-136c9.4-9.2 9.4-24.4 0-33.8z"
            />
          </svg>
        </button>
      </div>
    </nav>
  </div>
</DocumentFragment>
`;
